src/presetsController.ts

```diff
@@ -174,6 +175,11 @@ export class PresetsController {
         // Private fields must be set after validation, otherwise validation would fail.
         this.populatePrivatePresetsFields(presetsFile, file);
         await this.mergeIncludeFiles(presetsFile, presetsFile, file, referencedFiles);
+
+        // set the pre-expanded version so we can call expandPresetsFile on it
+        setPresetsFile(this.folderPath, presetsFile);
```

Comment:
What about the current code requires that we call `setPresetsFile` first? It seems to me that it's passed in as an argument to `expandPresetsFile`

Reply:
Its passed to `expandPresetsFile` so we can update it by reference, but `setPresetsFile` is still required because each of the individual `expand___Preset` functions dont take a whole presets file as a parameter - we call `getPresetByName(configurePresets(folder, useOriginalPresets), name)` or something similar which reads the preset file that we already have saved

Reply:
From my most recent change, we could do this same thing but read from original presets file instead of this semi processed version of expanded presets... though it would effectively be the same thing just a matter of it looking cleaner one way or the other